internal/envconfig/envconfig.go

```diff
@@ -12,168 +12,499 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package envconfig resolves configuration and secret values from environment
-// variables. This works by transforming the environment variables and then
-// invoking github.com/kelseyhightower/envconfig to map the environment
-// variables into a provided struct.
+// Package envconfig populates struct fields based on environment variable
+// values (or anything that responds to "Lookup"). Structs declare their
+// environment dependencies using the `env` tag with the key being the name of
+// the environment variable, case sensitive.
 //
-// If an environment variable begins with "secret://", the remaining string bits
-// are used to resolve the value in the provided SecretManager. For example:
+//     type MyStruct struct {
+//         A string `env:"A"` // resolves A to $A
+//         B string `env:"B,required"` // resolves B to $B, errors if $B is unset
+//         C string `env:"C,default=foo"` // resolves C to $C, defaults to "foo"
 //
-//     FOO=secret://foo/bar/baz => secretmanager.GetSecretValue("foo/bar/baz")
+//         D string `env:"D,required,default=foo"` // error, cannot be required and default
+//         E string `env:""` // error, must specify key
+//     }
 //
-// The environment variables are rewritten to be the secret value, but this is
-// only visible within the running process and any child processes.
+// All built-in types are supported except Func and Chan. If you need to define
+// a custom decoder, implement Decoder:
 //
-// If an environment variable secret ends with "?target=file" then the resulting
-// secret value is written to SECRETS_DIR and the environment variable is
-// updated to be the local path to that file.
+//     type MyStruct struct {
+//         field string
+//     }
+//
+//     func (v *MyStruct) EnvDecode(val string) error {
+//         v.field = fmt.Sprintf("PREFIX-%s", val)
+//         return nil
+//     }
+//
+// In the environment, slices are specified as comma-separated values:
+//
+//     export MYVAR="a,b,c,d" // []string{"a", "b", "c", "d"}
+//
+// In the environment, maps are specified as comma-separated key:value pairs:
+//
+//     export MYVAR="a:b,c:d" // map[string]string{"a":"b", "c":"d"}
+//
+// If you need to modify environment variable values before processing, you can
+// specify a custom mutator:
+//
+//     type Config struct {
+//         Password `env:"PASSWORD_SECRET"`
```

Comment:
i don't like that this isn't drop in replacement for the existing secret:// syntax

Reply:
ah - this is confusing, since we used this example but not how our secrets are specified :)

Reply:
The key here is that _we_ define how secrets are parsed in the secrets package. That was just an example. I moved it to its own package and I think that makes it less confusing now.